spec/dataframe_spec.rb

```diff
@@ -2372,4 +2658,213 @@ def create_test(*args, &_proc)
+
+      describe 'header' do
+        subject(:header) { table.at('tr:first-child > th:first-child') }
+        it { is_expected.not_to be_nil }
+        its(['colspan']) { is_expected.to eq (df.ncols + 1).to_s }
+        its(:text) { is_expected.to eq "Daru::DataFrame:#{df.object_id} rows: 3 cols: 3" }
+      end
+
+      describe 'column headers' do
+        subject(:name) { table.search('tr:nth-child(2) th').map(&:text) }
+        its(:size) { is_expected.to eq df.ncols + 1 }
+        it { is_expected.to eq ['', 'a', 'b', 'c'] }
+      end
+
+      describe 'index' do
+        subject(:indexes) { table.search('tr > td:first-child').map(&:text) }
+        its(:count) { is_expected.to eq df.nrows }
+        it { is_expected.to eq df.index.to_a.map(&:to_s) }
+      end
+
+      describe 'values' do
+        subject(:values) {
+          table.search('tr')[2..-1]
+               .map { |tr| tr.search('td')[1..-1].map(&:text) }
+        }
+        its(:count) { is_expected.to eq df.nrows }
+        it { is_expected.to eq df.map_rows{|r| r.map(&:to_s)} }
+      end
+    end
+
+    context 'large dataframe' do
+      let(:df) { Daru::DataFrame.new({a: [1,2,3]*100, b: [3,4,5]*100, c: [6,7,8]*100}, name: 'test') }
+
+      it 'has only 30 rows (+ 2 header rows, + 2 finishing rows)' do
+        expect(table.search('tr').size).to eq 34
+      end
+
+      describe '"skipped" row' do
+        subject(:row) { table.search('tr:nth-child(33) td').map(&:text) }
+        its(:count) { is_expected.to eq df.ncols + 1 }
+        it { is_expected.to all eq '...' }
+      end
+
+      describe 'last row' do
+        subject(:row) { table.search('tr:nth-child(34) td').map(&:text) }
+        its(:count) { is_expected.to eq df.ncols + 1 }
+        it { is_expected.to eq ['299', *df.row[-1].map(&:to_s)] }
+      end
+    end
+  end
+
+  context '#to_s' do
+    it 'produces something, despite of how reasonable you think it is' do
+      expect(@data_frame.to_s).to eq @data_frame.to_html
+    end
+  end
+
+  context '#to_json' do
+    let(:df) { Daru::DataFrame.new({a: [1,2,3], b: [3,4,5], c: [6,7,8]}, index: [:one, :two, :three], name: 'test')}
+    subject { JSON.parse(json) }
+
+    context 'with index' do
+      let(:json) { df.to_json(false) }
+      # FIXME: is it most reasonable we can do?.. -- zverok
+      # For me, more resonable thing would be something like
+      #
+      # [
+      #   {"index" => "one"  , "a"=>1, "b"=>3, "c"=>6},
+      #   {"index" => "two"  , "a"=>2, "b"=>4, "c"=>7},
+      #   {"index" => "three", "a"=>3, "b"=>5, "c"=>8}
+      # ]
+      #
+      # Or maybe
+      #
+      # [
+      #   ["one"  , {"a"=>1, "b"=>3, "c"=>6}],
+      #   ["two"  , {"a"=>2, "b"=>4, "c"=>7}],
+      #   ["three", {"a"=>3, "b"=>5, "c"=>8}]
+      # ]
+      #
+      # Or even
+      #
+      # {
+      #   "one"   => {"a"=>1, "b"=>3, "c"=>6},
+      #   "two"   => {"a"=>2, "b"=>4, "c"=>7},
+      #   "three" => {"a"=>3, "b"=>5, "c"=>8}
```

Comment:
In fact, I even think that the latter form:
```ruby
{
  "one"   => {"a"=>1, "b"=>3, "c"=>6},
  "two"   => {"a"=>2, "b"=>4, "c"=>7},
  "three" => {"a"=>3, "b"=>5, "c"=>8}
}
```

...is a pretty good candidate for dump/store (small) DataFrame in JSON. But I'm absolutely not insisting on focusing on this matter now, let's just leave (or, if you want, remove) this "FIXME" for future generations :)